pinot-common/src/main/java/com/linkedin/pinot/common/metadata/ZKMetadataProvider.java

```diff
@@ -142,6 +142,13 @@ public static void setInstancePartitionAssignmentFromPropertyStore(ZkHelixProper
         partitionMappingZKMetadata.toZNRecord(), AccessOption.PERSISTENT);
   }
 
+  public static boolean setOfflineSegmentZKMetadataOnlyIfCorrectVersion(ZkHelixPropertyStore<ZNRecord> propertyStore,
```

Comment:
Are you intending to use znode version here?

Reply:
yes, i pass in the version from offline segment zk metadata, where i set the znode version